### ISSUE TITLE
Have gRPC pass missing fields as NULLs instead of empty strings in PutObservations

### ISSUE THREAD

**Issue by jo-asplin-met-no** [EURODEO/e-soh#50]
In the `PutObservations` method, missing string fields are converted to empty strings instead of NULLs. This will thus essentially break the representation invariant for columns that are specified as NOT NULL in the database schema. We need to ensure that gRPC passes missing fields as NULLs so that the error can be detected by the database engine and thus reported back immediately to the client (which then needs to be fixed somehow, such as e.g. supplying dummy values for these fields).

**Comment by jo-asplin-met-no** [comment]
[This post](https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3) seems to be relevant. So for instance specify a field as `optional` to be able to distinguish default value from missing value?

**Comment by lukas-phaf** [comment]
Indeed, making non-required fields optional seems to be the way to go. The protobuf docs have a good description of this:
https://protobuf.dev/programming-guides/proto3/#field-labels

```
optional: An optional field is in one of two possible states:

the field is set, and contains a value that was explicitly set or parsed from the wire. It will be serialized to the wire.
the field is unset, and will return the default value. It will not be serialized to the wire.
You can check to see if the value was explicitly set.
```

I think this is exactly what we need. The ingestor should only set fields on the protobuf messages that it actually needs (don't set any empty strings). In the store, we should check for set versus unset, and map unset to NULL in the database. 

When the store gets data from the database, it should do the opposite, only actually set the protobuf field if the database value is not NULL.

How I read this the sizes on the wire for each field would be:
- unset: 0
- empty string: 2   -> we should hardly ever need this?
- actual string: string length + 2

**Comment by lukas-phaf** [comment]
I switched the locations endpoint over to using this.

For the KNMI test data, the response size before this change was `341012`. It took 96 ms.
The response size after the change is `220314`.

The size change is limited, but the KNMI dataset really only has the fields filled in to make the API work. It took 98 ms.
